src/lua/api-gateway/validation/factory.lua

```diff
@@ -101,9 +102,9 @@ local function _generateHmacSignature()
     return hmacSignatureValidator:generateSignature()
 end
 
-local function _validateOAuthToken()
+local function _validateOAuthToken(obj)
     local oauthTokenValidator = OAuthTokenValidator:new()
-    return oauthTokenValidator:validateRequest()
+    BaseValidator:exitFn(oauthTokenValidator:validateRequest(obj))
```

Comment:
True, we should keep consistency with the other validators.
The idea behind this change was that I may want to create a custom, composite validator out of existing built-in validators. So I may need to repeatedly call `validateRequest` on different validators, which would not be possible right now since `validateRequest` writes on the response. That's why I moved the code that writes on the response in the wrapper, so for a new composite validator I would just create a new wrapper.